components/raftstore-v2/src/operation/command/admin/split.rs

```diff
@@ -0,0 +1,884 @@
+            "region_id" => region_id,
+            "region" => ?region,
+        );
+
+        // todo(SpadeA)
+        // let mut replication_state =
+        // self.ctx.global_replication_state.lock().unwrap(); new_peer.peer.
+        // init_replication_mode(&mut replication_state);
+        // drop(replication_state);
+
+        for p in region.get_peers() {
+            self.insert_peer_cache(p.clone());
+        }
+
+        // New peer derive write flow from parent region,
+        // this will be used by balance write flow.
+        self.set_peer_stat(parent_stat);
+        let last_compacted_idx = self
+            .storage()
+            .apply_state()
+            .get_truncated_state()
+            .get_index()
+            + 1;
+        self.set_last_compacted_idx(last_compacted_idx);
+
+        let campaigned = self.maybe_campaign(parent_is_leader);
+        if campaigned {
+            self.set_has_ready();
+        }
+
+        if parent_is_leader {
+            self.set_approximate_size(approximate_size);
+            self.set_approximate_keys(approximate_keys);
+            *self.txn_ext().pessimistic_locks.write() = locks;
+            // The new peer is likely to become leader, send a heartbeat immediately to
+            // reduce client query miss.
+            self.heartbeat_pd(store_ctx);
+        }
+
+        let mut tablet = store_ctx
+            .tablet_factory
+            .open_tablet(
+                region_id,
+                Some(RAFT_INIT_LOG_INDEX),
+                OpenOptions::default().set_create(true),
+            )
+            .unwrap();
+        self.tablet_mut().set(tablet.clone());
+
+        meta.tablet_caches.insert(region_id, self.tablet().clone());
+        meta.regions.insert(region_id, region.clone());
+        let not_exist = meta
+            .region_ranges
+            .insert(enc_end_key(&region), region_id)
+            .is_none();
+        assert!(not_exist, "[region {}] should not exist", region_id);
+        meta.readers
+            .insert(region_id, self.generate_read_delegate());
+        meta.region_read_progress
+            .insert(region_id, self.read_progress().clone());
+
+        if !campaigned {
+            if let Some(msg) = meta
+                .pending_msgs
+                .swap_remove_front(|m| m.get_to_peer() == self.peer())
+            {
+                let peer_msg = PeerMsg::RaftMessage(Box::new(msg));
+                if let Err(e) = store_ctx.router.force_send(region_id, peer_msg) {
+                    warn!(self.logger, "handle first requset failed"; "region_id" => region_id, "error" => ?e);
+                }
+            }
+        }
+
+        drop(meta);
+
+        let ranges_to_delete = get_range_not_in_region(&region);
+        tablet
+            .delete_ranges_cfs(DeleteStrategy::DeleteFiles, &ranges_to_delete)
+            .unwrap_or_else(|e| {
+                error!(self.logger,"failed to delete files in range"; "err" => %e);
+            });
```

Comment:
> let ranges_to_delete = get_range_not_in_region(&region);
>         tablet
>             .delete_ranges_cfs(DeleteStrategy::DeleteFiles, &ranges_to_delete)
>             .unwrap_or_else(|e| {
>                 error!(self.logger,"failed to delete files in range"; "err" => %e);
>             });

There's no point to call it here which blocks the peer's serving requests.
Because it's still the best effort style and DeleteStrategy::DeleteFiles will not delete entries of the ranges_to_delete if they live in a file shared with other ranges.